llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -330,6 +330,18 @@ Instruction *InstCombinerImpl::visitMul(BinaryOperator &I) {
       return BinaryOperator::CreateMul(X, X);
   }
 
+  {
+    Value *X, *Y;
+    // abs(X) * abs(Y) -> abs(X * Y)
+    if (I.hasNoSignedWrap() &&
+        match(Op0, m_Intrinsic<Intrinsic::abs>(m_Value(X), m_AllOnes())) &&
+        match(Op1, m_Intrinsic<Intrinsic::abs>(m_Value(Y), m_AllOnes())))
```

Comment:
You will also need `m_OneUse`. Please also add multi-use tests.

Reply:
Done. Thanks!